src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Minigame.kt

```diff
@@ -0,0 +1,129 @@
+    "minigame",
+    CommandCategory.GAME,
+    "Play a fun quiz with your friends!",
+    "Run `minigame` to begin a new game, and react within the countdown to join.\nRun `minigame skip` to skip a question you do not wish to answer."
+) {
+    // String is the channel ID
+    private var minigames = mutableMapOf<String, MinigameInstance>()
+    private val listener = MinigameListener(this)
+    private val questions: List<Question>
+
+    init {
+        val json = jackson.readTree(this.javaClass.getResource("/minigame.json"))
+        questions = jackson.readValue(json.traverse(), object : TypeReference<List<Question>>() {})
+    }
+
+    override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
+        val id = event.channel.id
+
+        // Handling for when a game is already in progress
+        val minigame = minigames[id]
+        if (minigame != null) {
+            // If a user attempts to skip a question
+            if (args.getOrNull(0) == "skip") {
+                if (!minigame.players.contains(event.author)) {
+                    event.channel.sendMessage("You can't skip a question if you aren't in the game.").queue()
+                    return
+                }
+
+                if (minigame.state != State.IN_PROGRESS) {
+                    event.channel.sendMessage("The game has not started yet!").queue()
+                    return
+                }
+
+                minigame.progress(event)
+                return
+            }
+
+            // If the user attempts to start a new game while a game is already in progress,
+            // Either cancel it if it's stale or indicate that a game is already in progress
+            if (System.currentTimeMillis() - minigame.timer > 30_000) {
+                event.channel.sendMessage("Cancelling stale game...").queue()
+                unregister(event.jda, minigame)
+            } else {
+                event.channel.sendMessage("A game is already running!").queue()
+                return
+            }
+        }
+
+        // Register our listener if it doesn't already exist
+        if (!event.jda.registeredListeners.contains(listener)) {
+            event.jda.addEventListener(listener)
+        }
+
+        val maxRounds = args.getOrNull(0)?.toIntOrNull() ?: 7
+        if (maxRounds < 2 || maxRounds > 10) {
+            event.channel.sendMessage("The number of rounds has to be greater than 1 and less than 11.").queue()
+            return
+        }
+
+        minigames[id] = MinigameInstance(
+            questions.shuffled().toMutableList(),
+            event.channel.id,
+            this,
+            maxRounds
+        )
+
+        minigames[id]!!.start(event)
+    }
+
+    // Used by MinigameInstances to indicate that they're done
+    fun unregister(client: JDA, minigame: MinigameInstance) {
+        minigames.remove(minigame.id)
+
+        // If no games are running, there's no point in listening for events
```

Comment:
We will get these events regarding of the listener being registered

Reply:
So should I keep the listener registered even when no games are running?

Reply:
Yeah, that reduces the complexity by a lot as you don't have to keep track of that